solution.rb

```diff
@@ -0,0 +1,24 @@
+require 'nokogiri'
+require 'open-uri'
+
+# open from internet instead
+xml_file = URI.open('http://www.ibiblio.org/xml/examples/shakespeare/macbeth.xml')
+xml_doc = Nokogiri::XML(xml_file)
+
+character_lines = Hash.new(0)
+
+xml_doc.xpath('//SPEECH').each do |speech|
+    speaker = speech.at_xpath('SPEAKER').text
+    if speaker == "ALL"
+        next
+    end
```

Comment:
Ruby has a shortcut for this.

```suggestion
    next if speaker == "ALL"
```